src/GameFactory.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+
+class GameFactory
+{
+    /**
+     * @var Color[]
+     */
+    private $colors;
+
+    /**
+     * @param array $colorNames
+     */
+    public function __construct(array $colorNames)
+    {
+        if (count($colorNames) > 6) {
```

Comment:
Würde ich nicht so limitieren, auch nicht in construct tuhn, sondern eine configuration klasse injecten